packages/std/src/traits.rs

```diff
@@ -64,8 +64,18 @@ pub trait Storage {
 /// We can use feature flags to opt-in to non-essential methods
 /// for backwards compatibility in systems that don't have them all.
 pub trait Api {
-    fn canonical_address(&self, human: &HumanAddr) -> StdResult<CanonicalAddr>;
-    fn human_address(&self, canonical: &CanonicalAddr) -> StdResult<HumanAddr>;
+    /// Takes a human readable address and validates if it's correctly formatted.
+    fn addr_validate(&self, human: &HumanAddr) -> StdResult<()> {
```

Comment:
I like this.

I wonder if we should pull the mock implementation into cosmwasm-vm, so there is the wasm -> rust call in the contracts, and we can later look to optimize this in the rust->go side if so desired, without breaking the contract ABI (where older contracts would use this in-wasm implementation and new ones use the wasm import.

Reply:
Sure, no worries regarding the implementation. Right now we copy the canical address to the contract and then discard it, which is obviously more than necessary. We can do all that if we decide we want the overall direction.

Reply:
Yes, this is good enough for a PR. I would just like the other form before release (if we include it)